### PR TITLE
cleanup: remove warnings throughout the compiler

### PR DESCRIPTION
**Pull request by saem**
## Summary

- mostly cleaning up unused variables and procs
- consolidated max macro pragma and generic instantiation const

### CODE REVIEW

**Comment by saem** [comment]
~Not worth it.~

**Comment by saem** [comment]
welp.. this is a big change.

Don't need a detailed review but I blitzed through imports and took out a bunch of warnings.

@zerbina or @alaviss?

File: lib/nimrtl.nim [hunk 1]
```diff
@@ -12,8 +12,7 @@ batchable: false
 #
 
 ## Main file to generate a DLL from the standard library.
-## The default Nimrtl does not only contain the `system` module, but these
-## too:
+## The default Nimrtl does contains the `system` module as well as:
```
Comment (zerbina) [inline lib/nimrtl.nim]:
```suggestion
## The default Nimrtl does contain the `system` module as well as:
```

File: compiler/ast/lexer.nim [hunk 2]
```diff
@@ -748,12 +750,12 @@ proc getEscapedChar(L: var Lexer, tok: var Token) =
...
   template registerLine =
-    let col = L.getColNumber(pos)
-
     when not defined(nimpretty):
-      if col > MaxLineLength:
+      if L.getColNumber(pos) > MaxLineLength:
         L.localReportPos(
           LexerReport(kind: rlexLineTooLong), pos)
+    else:
+      discard
```
Comment (zerbina) [inline compiler/ast/lexer.nim]: I think omitting the empty else clause in this case improves readability. Just a suggestion - no change expected from my side.

**Review by zerbina — COMMENTED** [review]
Found a small typo, but otherwise, looks good!

File: lib/nimrtl.nim (hunk 1 above)
Comment (saem) [reply lib/nimrtl.nim]: Thanks.

File: compiler/ast/lexer.nim (hunk 2 above)
Comment (saem) [reply compiler/ast/lexer.nim]: Fair enough, removed.

**Comment by saem** [comment]
Bors r+

**Pull request merged**